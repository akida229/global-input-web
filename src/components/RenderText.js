import React, {Component} from 'react'


export default class RenderText extends Component{
      render(){
        if(this.props.content){
          return(
            <div className={this.props.containerClass}>
              <h2 className={this.props.titleClass}>
                    {this.props.title}
              </h2>
              {this.props.content.map((p,index)=>{
                  return(<h3 className={this.props.contentClass} key={index}>{p}</h3>);
                 })
              }
            </div>
          );
        }
        else{
          return null;
        }


      }
}
