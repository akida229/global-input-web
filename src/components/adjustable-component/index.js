import React from 'react';
import {screenMedia} from '../screen-media';
export default class ResizableComponent extends React.Component{

      constructor(props){
        super(props);
        this.onWindowResize=this.onWindowResize.bind(this);
      }
      isScreenBiggerThan(size){
        return screenMedia.biggerThan(size);
      }
       componentDidMount() {
           window.addEventListener("resize", this.onWindowResize);
           this.scrollTo(this.props.scrollTo);
       }
       scrollTo(elementId){
            var elmnt = document.getElementById(elementId);
            if(elmnt){
                     window.scrollBy({top: -70,behavior: "smooth"});
                     elmnt.scrollIntoView();
            }
       }
       componentWillUnmount() {
           window.removeEventListener("resize", this.onWindowResize);
       }
       onWindowResize(){

          this.forceUpdate();
       }

}
