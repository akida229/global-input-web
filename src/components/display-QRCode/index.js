import DisplayQRCode from "./DisplayQRCode";
export default DisplayQRCode;
