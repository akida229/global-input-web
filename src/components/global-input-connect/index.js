import GlobalInputConnect from './GlobalInputConnect';
export default GlobalInputConnect;
