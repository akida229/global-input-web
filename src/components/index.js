
import BookMark from "./bookmark/BookMark";
import DisplayTextImage from "./text-image/DisplayTextImage";
import DisplayVideo from "./text-image/DisplayVideo";
import NotificationMessage from "./text-image/NotificationMessage";
import DisplayStaticContent from "./text-image/DisplayStaticContent";
import DisplaySourceCode from "./text-image/DisplaySourceCode";
import DisplaySectionTitle from "./text-image/DisplaySectionTitle";
import TextViewPad from "./text-image/TextViewPad";
import ShowHideButton from "./input/ShowHideButton";
import InputWithLabel from "./input/InputWithLabel";
import InputWithSelect from "./input/InputWithSelect";
import TextAreaWithSelect from "./input/TextAreaWithSelect";
import TextAreaWithLabel from "./input/TextAreaWithLabel";
import TextButton         from "./input/TextButton";
import ClipboardButton from "./input/ClipboardButton";
import TextRadioButtons from "./input/TextRadioButtons";
import TextSelectOptions from "./input/TextSelectOptions";
import TopMenu from "./top-menu";

export {TopMenu,DisplayTextImage,DisplayVideo,DisplayStaticContent,DisplaySourceCode,DisplaySectionTitle,TextViewPad,
        ShowHideButton,
        InputWithLabel,InputWithSelect,TextSelectOptions,TextAreaWithSelect,TextAreaWithLabel,TextButton,ClipboardButton,TextRadioButtons,NotificationMessage,BookMark};
