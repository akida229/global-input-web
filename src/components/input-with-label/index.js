import React, {Component} from 'react';
import "bootstrap/dist/css/bootstrap.min.css";
import PropTypes from 'prop-types';
import {styles} from "./styles";

export default class InputWithLabel extends Component{
  render(){
      
         return(
           <div className="form-group" style={this.props.fieldContainer}>
                   <input  className="form-control" id={this.props.id}
                   type={this.props.type}
                   min={this.props.min}
                   style={this.props.style}
                   max={this.props.max}
                   step={this.props.step}
                   readOnly={this.props.readOnly}
                   onKeyUp={evt=>{
                     if(this.props.onKeyEnter && evt && evt.keyCode===13){
                        this.props.onKeyEnter();
                     }
                   }}
                   onChange={(evt) => {
                     if(this.props.onChange){
                        this.props.onChange(evt.target.value,this.props.id);
                     }

                 }} value={this.props.value} required/>
               <label htmlFor={this.props.id} className="form-control-placeholder" onClick={this.labelClicked.bind(this)}>
                    <span style={this.props.labelStyle}>{this.props.label}</span>

                 </label>
                 {this.renderHelp()}
           </div>
         );
  }
  renderLabel(){
    if(this.props.label){

    }
  }
  renderHelp(){
      if(this.props.help){

        return(
          <div style={this.props.helpStyle}>{this.props.help}</div>
          );
      }
      else{
        return null;
      }

  }
  labelClicked(){
      document.getElementById(this.props.id).focus();

  }

}


InputWithLabel.propTypes={
    id:PropTypes.string.isRequired,
    type:PropTypes.string,
    label:PropTypes.string,
    readOnly:PropTypes.bool,
    onKeyEnter:PropTypes.func,
    onChange:PropTypes.func,
    style:PropTypes.object,
    labelStyle:PropTypes.object,
    help:PropTypes.string,
    helpStyle:PropTypes.object
}
InputWithLabel.defaultProps={
    type:"text",


}
