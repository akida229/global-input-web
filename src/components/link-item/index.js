import React from 'react';
import {Link} from 'react-router-dom';
export default class LinkItem extends React.Component{

    constructor(props){
      super(props);
      this.state={
        hover:false
      }
    }
    onMouseOver(){
        this.setState({hover:true});
    }
    onMouseOut(){
      this.setState({hover:false});
    }
    render(){
      let image=this.props.image;
      if(this.state.hover){
            image=this.props.imageHover;
      }
      if(this.props.to){
        return(
          <Link to={this.props.to} onMouseOver= {this.onMouseOver.bind(this)}
          onMouseOut = {this.onMouseOut.bind(this)}>
              <img  style={this.props.imageStyle}
                    src={image}/>
              {this.props.children}
          </Link>
        );
      }
      else{


          return(
            <a href={this.props.href} onMouseOver= {this.onMouseOver.bind(this)}
            onMouseOut = {this.onMouseOut.bind(this)} target="_blank">
                <img  style={this.props.imageStyle}
                      src={image}/>
                {this.props.children}
            </a>
          );
      }
    }


}
