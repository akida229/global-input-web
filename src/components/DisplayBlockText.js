import React, {Component} from 'react'
import "bootstrap/dist/css/bootstrap.min.css";
import "./styles/index.css";
import {globalStyles} from "./styles";
import RenderText from "./RenderText";

import ReactMarkdown from "react-markdown";



export default class DisplayBlockText extends Component{
      render(){
        if(this.props.content){
          return(
            <div className="row top-margin-md">
              <h2 className="section-heading">
                    {this.props.title}
              </h2>
              {this.props.content.map((p,index)=>{
                  return(<h3 className="section-subheading text-muted bottom-margin-sm" key={index}>
                      <ReactMarkdown source={p}/>
                  </h3>);
                 })
              }
            </div>
          );
        }
        else{
          return null;
        }


      }
}
