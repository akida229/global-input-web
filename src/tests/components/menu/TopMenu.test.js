import React from 'react';
import "../../setup";
import {TopMenu} from "../../../components";
import renderer from 'react-test-renderer';
import {render, fireEvent, wait} from 'react-testing-library';

var menus=[{link:"/", linkText:"home"}];
var decodeHTML=function(str) {
			return str.replace(/&#(\d+);/g, function(match, dec) {
				return String.fromCharCode(dec);
			});
		};

test("Test by snapshot",()=>{

    const topMenu=renderer.create(
      <TopMenu menus={menus}/>
    );
    let tree=topMenu.toJSON();
    expect(tree).toMatchSnapshot();
});
test("Test by html",()=>{
    const {getByText,getByTestId,container}=render(
      <TopMenu menus={menus}/>
    );
    const elem=getByTestId('mobile-to-open-menu');
    expect(elem.innerHTML).toBe(decodeHTML('&#9776;'));

});
