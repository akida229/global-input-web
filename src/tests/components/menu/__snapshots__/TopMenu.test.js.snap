// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test by snapshot 1`] = `
<div
  style={
    Object {
      "alignItems": "flex-start",
      "display": "flex",
      "flexDirection": "column",
      "justifyContent": "flex-start",
      "left": 0,
      "position": "fixed",
      "top": 0,
      "width": "100%",
      "zIndex": 100,
    }
  }
>
  <div
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#EEEEEE",
        "borderBottomColor": "#CCCCCC",
        "borderBottomStyle": "solid",
        "display": "flex",
        "elevation": 10,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingRight": 30,
        "paddingTop": 10,
        "shadowColor": "#888888",
        "shadowOffset": Object {
          "height": 10,
          "width": 10,
        },
        "shadowOpacity": 0.75,
        "shadowRadius": 5,
        "width": "100%",
      }
    }
  >
    <img
      style={
        Object {
          "marginBottom": 4,
          "marginLeft": 20,
          "maxWidth": 80,
        }
      }
    />
    <div
      style={
        Object {
          "marginLeft": 10,
          "paddingTop": 5,
        }
      }
    >
      <div
        style={
          Object {
            "color": "#4880ED",
            "fontFamily": "Elysio-Light, Helvetica, Arial, sans-serif",
            "fontSize": 20,
            "fontWeight": 300,
            "whiteSpace": "nowrap",
          }
        }
      />
    </div>
    <div
      style={Object {}}
    >
      <a
        data-testid="mobile-to-open-menu"
        href="#b"
        onClick={[Function]}
        style={
          Object {
            "color": "#4880ED",
            "fontSize": 40,
            "fontStyle": "normal",
            "fontWeight": "normal",
          }
        }
      >
        â˜°
      </a>
    </div>
  </div>
</div>
`;
